lib/midcom/helper/imagepopup/handler/upload.php

```diff
@@ -0,0 +1,161 @@
+<?php 
+
+/**
+ * This handler uploads the attachments attached to object $object.
+ *
+ * @package midcom.helper.imagepopup
+ */
+
+class midcom_helper_imagepopup_handler_upload extends midcom_baseclasses_components_handler
+{	
+	public function _handler_upload($handler_id, array $args, array &$data)
+	{
+		// Ensure we get the correct styles
+		midcom::get()->style->prepend_component_styledir('midcom.helper.imagepopup');
+		midcom::get()->skip_page_style = true;
+		
+		
+		// Get the file
+		reset ($_FILES);
+		$temp = current($_FILES);
+		
+		$accepted_origins = array("http://localhost", "http://openpsa/");
```

Comment:
This isn't useful for real hosting, please remove or replace with a version that uses the actual hostname. Maybe this should be a configuration option